File: in/benchmark.m
% Benchmark

  dx = 100.;
  dt = .0075;
  faultnormal = 0;
  fixhypo = 2;
  rsource = 50.;
  debug = 0;
  oplevel = 6;
  oplevel = 5;
  oplevel = 4;
  oplevel = 3;
  oplevel = 2;
  oplevel = 1;
  hourglass = [ 1. 1. ];
  bc1 = [ 1 0 0 ];
  npml = 1;

  % 4^3
  nt = 4;
  nn = [ 16 16 16 ]; np = [ 8 8 8 ];
  nn = [  8  8  8 ]; np = [ 4 4 4 ];
  nn = [  4  4  4 ]; np = [ 2 2 2 ];
  nn = [  2  2  2 ]; np = [ 1 1 1 ];

  % 128^3
  nt = 128;
  nn = [ 1024 1024 1024 ]; np = [ 8 8 8 ];
  nn = [  512  512  512 ]; np = [ 4 4 4 ];
  nn = [  256  256  256 ]; np = [ 2 2 2 ];
  nn = [  128  128  128 ]; np = [ 1 1 1 ];

  % 32^3
  nt = 32;
  nn = [ 256 256 256 ]; np = [ 8 8 8 ];
  nn = [ 128 128 128 ]; np = [ 4 4 4 ];
  nn = [  64  64  64 ]; np = [ 2 2 2 ];
  nn = [  32  32  32 ]; np = [ 1 1 1 ];

  % 32^3
  nt = 64;
  nn = [ 512 512 512 ]; np = [ 8 8 8 ];
  nn = [ 256 256 256 ]; np = [ 4 4 4 ];
  nn = [ 128 128 128 ]; np = [ 2 2 2 ];
  nn = [  64  64  64 ]; np = [ 1 1 1 ];

File: tests/benchmark.py
#!/usr/bin/env python
"""
Benchmarks
"""

import sord, math

_np3 = [ (1,1,1) ]
_np3 = [ (1,1,1), (1,2,2), (1,4,4), (1,8,8), (1,16,16), (1,32,32), (1,64,64) ]
_np3 = [ (1,1,1), (1,2,2), (1,4,4) ]
_np3 = [ (1,1,1), (1,1,2) ]
#machine = 'tacc-ranger'; queue = 'large'; _np3 = [ (1,128,96), (1,128,128) ]
optimize = 'O'
optimize = 'p'
mode = 's'
mode = 'm'
_n = 20
_n = 200
nt = 8
nt = 16
itstats = 2
itstats = 1
oplevel = 2
oplevel = 6

itcheck = -1
itio = nt
debug = 0
dx = 100.
dt = 0.0075
bc1 = bc2 = 0, 0, 0
npml = 0
hourglass = 1., 1.
faultnormal = 0
fieldio = [
    ( '=', 'rho', [],     2670.0 ),      
    ( '=', 'vp',  [],     6000.0 ),      
    ( '=', 'vs',  [],     3464.0 ),      
    ( '=', 'gam', [],        0.0 ),      
    ( '=s', 'v1', [0,0,0,1], 1.0 ),
    ( '=s', 'v2', [0,0,0,1], 1.0 ),
    ( '=s', 'v3', [0,0,0,1], 1.0 ),
]

for np3 in _np3:
    nn = [ _n * _np for _np in np3 ]
    _np = np3[0] * np3[1] * np3[2]
    print '\nBenchmark: %s, %s, %s, %s' % ( np3, math.log(_np,4), _np/16, _np )
    sord.run( locals() )

